### ISSUE TITLE
Fairplay and Video.js not work with Safari 15 fro iPad and iPhone

### ISSUE THREAD

**Issue by argentinaluiz** [videojs/videojs-contrib-eme#140]
Hi there,

This week I tried videojs-eme on the new Safari 15 using iPhone or iPad and it didn't work.

The feedback says the content is not supported, but version 14 works.

**Comment by gkatsev** [comment]
I think it's broken on Apple's side.

**Comment by argentinaluiz** [comment]
Yeah, I tested a video with pure DRM without videojs and it worked.

**Comment by HankBrown** [comment]
Major fail at our website now (all our stuff is encrypted).
iPhone 7 iOS 15
https://developers.apple.com/forums/thread/690429 says to disable GPU Process in Safari/Advanced but no joy.

**Comment by BakurL** [comment]
Any idea on how to fix this?

**Comment by BakurL** [comment]
Workaround  perhaps?

**Comment by darcyrailip** [comment]
Is Apple even aware? This seems critical, we have a ton of clients complaining.

**Comment by eban5** [comment]
What is the expected error Safari 15 / Apple error happening here? I'm intermittently seeing either:

* **CODE 4**:  `The media could not be loaded, either because the server or network failed or because the format is not supported.` or 
* A few warnings in the console trying different m3u8s with `Problem encountered with playlist 3-hls-drm_4.m3u8. Could not determine codecs for playlist. Switching to playlist 0-hls-drm_1.m3u8.` then ultimately failing with **CODE 3**  MEDIA_ERR_DECODE `Playback cannot continue. No available working or supported playlists.`

For us, we noticed that native captions weren't populating in the captions or subtitles menus, so those icons never appear in the control bar. If we toggle on `Prefer Closed Captions` in macOS > System Preferences > Accessibility > Captions menu, then we do see the captions appear on screen. 

In-stream captions are defined in the m3u8 as below, but we don't see why it wouldn't be able to parse them and whether or not it is related to the above: 

```
#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="subtitles",NAME="Captions",DEFAULT=YES,AUTOSELECT=YES,FORCED=NO,LANGUAGE="eng",URI="hls-drm_5_0.m3u8"
```

**Comment by HankBrown** [comment]
"The error didn't come from video.js EME extension. To reproduce you can try implement FairPlay using pure JavaScript or using FPS_in_Safari_Example.html that come with FairPlay Streaming Server SDK. You will get the same error on iOS15. — nonth 7 hours ago"  [](https://developer.apple.com/forums/thread/690429)

**Comment by gkatsev** [comment]
By default, we still use the legacy standard for EME on Safari. Supposedly, the issue doesn't occur with the latest standards based EME API. I've been playing around locally and, unfortunately, haven't been able to get it working.
We do have this PR, which is a bit stale, but may help out: https://github.com/videojs/videojs-contrib-eme/pull/87

**Comment by BakurL** [comment]
FYI - Disable GPU rendering (GPU process: Media) in the Safari Experimental Settings, and restart Safari, all the players start showing video - comment from: https://developer.apple.com/forums/thread/690429 . This worked for non-videojs player

**Comment by gkatsev** [comment]
@BakurL yeah, we see some issues on our end as well. I think ultimately, we'll need to wait for apple for the remaining issue. Also, sorry that you'll need to make some more changes once 4.0 comes out. Hopefully, it'll allow you to simplify things on your end. See https://github.com/videojs/videojs-contrib-eme/pull/147, which still needs some reviews by my teammates, but it's working for me.

**Comment by yudarksoul** [comment]
demo videojs work on safari
https://codepen.io/yuchichang/pen/gOqRjLB
